gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java

```diff
@@ -135,7 +142,9 @@ public String toString() {
     }
   }
 
-  private BlockingQueue<Dag<JobExecutionPlan>> queue;
+  private LinkedBlockingDeque<Dag<JobExecutionPlan>>[] queue;
```

Comment:
updated the javadoc with the explanation. cancellation requests has to go to the same dagmanagerthread which accpeted the addSpec request, because that thread only knows the azkaban execution id